### PR TITLE
[UI] Align UI to the schema & API refactor - table and detail page

### PR DESCRIPTION
**Pull request by zhongnansu**
*Issue #, if available:*

*Description of changes:*

- Apply schema & API changes to UI - table and detail page
    - landing page image
    - report detail 
image
    - report definition detail 
image

- Link the newly-added `generate report by id` API to enable the click-on icon download  image

- Linked snapshot of dashboard/visualization to the report source field. It will open an new tab
image

- remove unused imports, optimize the table data access logic.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kibana-reports/public/components/main/report_definition_details/report_definition_details.tsx [hunk 1]
```diff
@@ -56,42 +57,50 @@ export function ReportDefinitionDetails(props) {
...
-      triggerType: data['trigger']['trigger_type'],
-      scheduleDetails: data['trigger']['trigger_params']['schedule_type'],
+      triggerType: trigger.trigger_type,
+      scheduleDetails: triggerParams ? triggerParams.schedule_type : `\u2014`,
       alertDetails: `\u2014`,
-      status: data['status'],
-      deliveryChannels: data['delivery']['channel'],
-      kibanaRecipients: data['delivery']['delivery_params']['recipients'],
+      status: reportDefinition.status,
+      deliveryChannels: `\u2014`,
```
Comment (akbhatta) [inline kibana-reports/public/components/main/report_definition_details/report_definition_details.tsx:98]: nit: may be also refactor `\u2014` (can be in another PR as well)

**Review by akbhatta — APPROVED** [review]

**Review by davidcui-amzn — APPROVED** [review]
